pynest/nest/tests/test_tsodyks2_connection.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+#
+# test_tsodyks2_connection.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import nest
+import unittest
+
+
+@nest.ll_api.check_stack
+class Tsodyks2ConnectionTest(unittest.TestCase):
+    """
+    Functional test for the "tsodyks2" synapse: compare NEST implementation to
+    a reference generated in the method reproduce_weight_drift(), for a
+    sequence of spike times coming from a Poisson generator.
+    """
+
+    def setUp(self):
+        self.resolution = 0.1  # [ms]
+        self.presynaptic_firing_rate = 20.0  # [Hz]
+        self.simulation_duration = 1E3  # [ms]
+        self.hardcoded_trains_length = 15.  # [ms]
+        self.synapse_parameters = {
+            "receptor_type": 1,
+            "delay": self.resolution,
+            "U": 1.0,
+            "u": 1.0,
+            "x": 1.0,
+            "tau_rec": 100.,
+            "tau_fac": 0.,
+            "weight": 1.  # maximal possible response (absolute synaptic efficacy)
+        }
+
+    def test_tsodyk2_synapse(self):
+        synapse_model = "tsodyks2_synapse_rec"
+        self.synapse_parameters["synapse_model"] = synapse_model
+
+        pre_spikes, weight_by_nest = self.do_the_nest_simulation()
+        weight_reproduced_independently = self.reproduce_weight_drift(
+            pre_spikes,
+            absolute_weight=self.synapse_parameters["weight"])
+
+        np.testing.assert_allclose(weight_reproduced_independently, weight_by_nest, atol=1E-12)
+
+    def do_the_nest_simulation(self):
+        """
+        This function is where calls to NEST reside.
+        Returns the generated pre- and post spike sequences
+        and the resulting weight established by the tsodyks2 synapse.
+        """
+        nest.set_verbosity('M_WARNING')
+        nest.ResetKernel()
+        nest.SetKernelStatus({'resolution': self.resolution})
```

Comment:
We should be consistent with using `'` or `"` for strings. Sorry I didn't see it sooner. `"` is mostly used here, so I've gone for that in my suggestions.
```suggestion
        nest.set_verbosity("M_WARNING")
        nest.ResetKernel()
        nest.SetKernelStatus({"resolution": self.resolution})
```